RecoTracker/TkTrackingRegions/plugins/AreaSeededTrackingRegionsBuilder.cc

```diff
@@ -0,0 +1,245 @@
+
+    // some common variables
+    const float x_rmin_phimin = area.x_rmin_phimin()-orig.x();
+    const float x_rmin_phimax = area.x_rmin_phimax()-orig.x();
+    const float y_rmin_phimin = area.y_rmin_phimin()-orig.y();
+    const float y_rmin_phimax = area.y_rmin_phimax()-orig.y();
+
+    const std::array<float, 2> p_rmin_phimin = {{x_rmin_phimin, y_rmin_phimin}};
+    const std::array<float, 2> p_rmin_phimax = {{x_rmin_phimax, y_rmin_phimax}};
+    const std::array<float, 2> p_rmax_phimin = {{area.x_rmax_phimin() - orig.x(), area.y_rmax_phimin() - orig.y()}};
+    const std::array<float, 2> p_rmax_phimax = {{area.x_rmax_phimax() - orig.x(), area.y_rmax_phimax() - orig.y()}};
+
+
+    // eta
+    {
+      // find which of the two rmin points is closer to the origin
+      //
+      // closest point for p_rmin, farthest point for p_rmax
+      const std::array<float, 2> p_rmin = perp2(p_rmin_phimin) < perp2(p_rmin_phimax) ? p_rmin_phimin : p_rmin_phimax;
+      const std::array<float, 2> p_rmax = perp2(p_rmax_phimin) > perp2(p_rmax_phimax) ? p_rmax_phimin : p_rmax_phimax;
+
+      // then calculate the xy vector from the point closest to p_rmin on
+      // the (origin,originRadius) circle to the p_rmin
+      // this will maximize the eta window
+      const auto p_min = vecFromOrigPlusRadius(p_rmin);
+      const auto p_max = vecFromOrigPlusRadius(p_rmax);
+
+      // then we calculate the maximal eta window
+      const auto etamin = std::min(etaFromXYZ(p_min[0], p_min[1], area.zmin() - (orig.z()+origin.second)),
+                                   etaFromXYZ(p_max[0], p_max[1], area.zmin() - (orig.z()+origin.second)));
+      const auto etamax = std::max(etaFromXYZ(p_min[0], p_min[1], area.zmax() - (orig.z()-origin.second)),
+                                   etaFromXYZ(p_max[0], p_max[1], area.zmax() - (orig.z()-origin.second)));
+
+      LogTrace("AreaSeededTrackingRegionBuilder") << "  eta min,max " << etamin << "," << etamax;
+
+      minEta = std::min(minEta, etamin);
+      maxEta = std::max(maxEta, etamax);
+    }
+
+    // phi
+    {
+      // For phi we construct the tangent lines of (origin,
+      // originRadius) that go though eahc of the 4 points (in xy
+      // plane) of the area. Easiest is to make a vector orthogonal to
+      // origin->point vector which has a length of
+      //
+      // length = r*d/std::sqrt(d*d-r*r)
+      //
+      // where r is the originRadius and d is the distance from origin
+      // to the point (to derive draw the situation and start with
+      // definitions of sin/cos of one of the angles of the
+      // right-angled triangle.
+
+      // but we start with a "reference phi" so that we can easily
+      // decide which phi is the largest/smallest without having too
+      // much of headache with the wrapping
+      const auto phi_ref = std::atan2(0.5f*(y_rmin_phimin + y_rmin_phimax),
+                                      0.5f*(x_rmin_phimin + y_rmin_phimax));
```

Comment:
should the second argument be with *x*? ``` 0.5f*(x_rmin_phimin + x_rmin_phimax) ```.
Also, 0.5f appears to be redundant here.

Reply:
Before I spend more time looking at the comparisons, please clarify if this is a bug that would affect performance.

Reply:
I agree, thanks for catching (I'm running MTV to test the effect).

Reply:
It can affect, so better to wait for an update (hopefully later today).